src/Frame.js

```diff
@@ -0,0 +1,96 @@
+function Frame() {
+  this.pins = 10
+  this.currentFrame = []
+  this.frameScores = []
+  this.turnCount = 0
+  this.spare = false
+  this.strike = false
+  this.game_over = false
+  this.points = 0
+}
+
+Frame.prototype.bowl = function() {
+  var x = (Math.round(Math.random() * this.pins ))
+  this.points = x
+  this.pins -= x
+  this.turnCount += 1
+    if (this.strike){
+        this.frameScores.slice(-1)[0].push(x)
+      if (this.frameScores.length >= 2) {
+      if (this.turnCount === 1)
+        {this.frameScores.slice(-2)[0].push(x)}}
+      this.currentFrame.push(x)
+      this.strike_count += 1
+      if (this.strike_count === 2){this.strike = false}
+    } else if (this.spare && this.turnCount === 1){
+      this.frameScores.slice(-1)[0].push(x)
+      this.currentFrame.push(x)
+      this.spare = false
+    } else { this.currentFrame.push(x) }
+  if(this.pins === 0){this._isStrikeorSpare()}
+  if (this.turnCount === 2) {
+    this.resetFrame();
+  }
+}
+
+Frame.prototype.resetFrame = function() {
+  this.turnCount = 0
+  this.pins = 10
+  this.frameScores.push(this.currentFrame)
+  this.currentFrame = []
+};
+
+Frame.prototype._isStrikeorSpare = function() {
+    if (this.currentFrame.length === 1) {this.strike = true
+    this.resetFrame()
+    this.strike_count = 0
+  } else {this.spare = true}
+}
+
+Frame.prototype.lastFrame_bowl = function() {
```

Comment:
This function has too many statements. (23)<br>This function's cyclomatic complexity is too high. (9)